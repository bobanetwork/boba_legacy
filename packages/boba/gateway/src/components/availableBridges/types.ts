export interface IBridges {
  name: string
  type: string
  link: string
  tokens: string[]
}

export interface AvailableBridgesProps {
  token?: any | null // FIXME: fix the type of token
  walletAddress: string
}
