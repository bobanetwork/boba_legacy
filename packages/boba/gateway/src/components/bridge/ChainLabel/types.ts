export interface ChainLabelInterface {
  direction: string
}

export type IconType = {
  ethereum: ({ selected }: { selected?: boolean | undefined }) => Element
  bnb: ({ selected }: { selected?: boolean | undefined }) => Element
  avax: ({ selected }: { selected?: boolean | undefined }) => Element
}

export type DirectionType = {
  from: JSX.Element
  to: JSX.Element
}
