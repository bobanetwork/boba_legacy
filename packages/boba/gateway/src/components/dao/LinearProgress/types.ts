export interface LinearProgressProps {
  A: number
  B: number
  C: number
}
