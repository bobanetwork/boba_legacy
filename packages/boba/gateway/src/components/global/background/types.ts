export enum BackgroundPosition {
  TOP = 'top',
  CENTER = 'center',
}

export interface BackgroundProps {
  position: BackgroundPosition
}

export const RoutesWithBackgroundPositionAtTop = ['/bridge', '/']
