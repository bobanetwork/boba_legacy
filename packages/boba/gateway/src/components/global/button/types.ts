import { ReactNode } from 'react'

type ButtonType = {
  disable?: boolean
  loading?: boolean
  small?: boolean
  outline?: boolean
  tiny?: boolean
  transparent?: boolean
}

export interface ButtonTypes extends ButtonType {
  label: ReactNode | string
  style?: Record<string, string>
  className?: string
  onClick?: () => void
}
