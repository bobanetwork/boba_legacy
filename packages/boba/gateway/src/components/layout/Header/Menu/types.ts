export interface IMenuItem {
  label: string
  path: string
}

export interface MenuProps {
  isOpen?: boolean
}
