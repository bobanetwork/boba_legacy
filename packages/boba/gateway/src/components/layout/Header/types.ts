export interface HeaderProps {}

export enum THEME_NAME {
  LIGHT = 'light',
  DARK = 'dark',
}
