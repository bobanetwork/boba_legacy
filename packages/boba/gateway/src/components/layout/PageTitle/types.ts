export interface PageTitleTypes {
  title: string
  slug: string
}
