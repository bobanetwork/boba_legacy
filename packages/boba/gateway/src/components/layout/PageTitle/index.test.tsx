import React from 'react'
import { BrowserRouter } from 'react-router-dom'
import { Provider } from 'react-redux'
import configureStore from 'redux-mock-store'

import { render } from '@testing-library/react'

import CustomThemeProvider from 'themes'
import { PageTitleTypes } from './types'

import { PageTitle } from '.'

const mockStore = configureStore()

const renderTypography = (props: PageTitleTypes) => {
  return render(
    <Provider
      store={mockStore({
        ui: {
          theme: 'dark',
        },
      })}
    >
      <BrowserRouter>
        <CustomThemeProvider>
          <PageTitle {...props}>Text goes here</PageTitle>
        </CustomThemeProvider>
      </BrowserRouter>
    </Provider>
  )
}

describe('PageTitle', () => {
  test('should match snapshots head', () => {
    const { asFragment } = renderTypography({
      title: 'Page Title',
      slug: 'This is the current slug',
    })
    expect(asFragment()).toMatchSnapshot()
  })
})
