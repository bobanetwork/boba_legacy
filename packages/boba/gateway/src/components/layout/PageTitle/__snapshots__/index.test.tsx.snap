// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageTitle should match snapshots head 1`] = `
<DocumentFragment>
  <div
    class="sc-ewnqHT fPumzW"
  >
    <h1
      class="sc-bgqQcB jYvPsJ sc-fFGjHI MzdwS"
    >
      Page Title
    </h1>
    This is the current slug
  </div>
</DocumentFragment>
`;
